packages/mdc-select/README.md

```diff
@@ -235,21 +348,40 @@ If you are using a JavaScript framework, such as React or Angular, you can creat
-| `hasLabel() => boolean` | Returns true if the `select` has a label associated with it. |
 | `floatLabel(value: boolean) => void` | Floats or defloats label. |
 | `getLabelWidth() => number` | Returns the offsetWidth of the label element. |
 | `hasOutline() => boolean` | Returns true if the `select` has the notched outline element. |
 | `notchOutline(labelWidth: number, isRtl, boolean) => void` | Switches the notched outline element to its "notched state." |
 | `closeOutline() => void` | Switches the notched outline element to its closed state. |
+| `openMenu() => void` | Causes the menu element in the enhanced select to open. |
+| `closeMenu() => void` | Causes the menu element in the enhanced select to close. |
+| `setValue(value: string) => void` | Sets the value of the select or text content of the selected-text element. |
+| `isMenuOpen() => boolean` | Returns true if the menu is currently opened. |
+| `setSelectedIndex(index: number) => void` | Selects the option or list item at the specified index. |
+| `setDisabled(isDisabled: boolean) => void` | Enables or disables the native or enhanced select. |
+| `setRippleCenter(normalizedX: number) => void` | Sets the line ripple center to the provided normalizedX value. |
+| `notifyChange({value: string}: Object) => void` | Emits the `MDCSelect:change` event when an element is selected. |
 
 ### `MDCSelectFoundation`
 
 | Method Signature | Description |
 | --- | --- |
 | `notchOutline(openNotch: boolean) => void` | Opens/closes the notched outline. |
-| `updateDisabledStyle(disabled: boolean) => void` | Updates appearance based on disabled state. This must be called whenever the `disabled` state changes. |
+| `setDisabled(isDisabled: boolean) => void` | Updates appearance based on disabled state. This must be called whenever the `disabled` state changes. |
 | `handleFocus() => void` | Handles a focus event on the `select` element. |
 | `handleBlur() => void` | Handles a blur event on the `select` element. |
+| `handleClick(normalizedX: number) => void` | Sets the line ripple center to the normalizedX for the line ripple. |
 | `handleChange() => void` | Handles a change to the `select` element's value. This must be called both for `change` events and programmatic changes requested via the component API. |
+| `handleKeydown(event: Event) => void` | Handles opening the menu (enhanced select) when the `mdc-select__selected-text` element is focused and the user presses the `Enter` or `Space` key. |
+| `setSelectedIndex(index: number) => void` | Handles setting the `mdc-select__selected-text` element and closing the menu (enhanced select only). Also causes the label to float and outline to notch if needed. |
+| `setValue(value: string) => void` | Handles setting the value through the adapter and causes the label to float and outline to notch if needed. |
+| `getValue() => string` | Handles getting the value through the adapter. |
```

Comment:
Should this not be public on the foundation? It seems like the foundation only needs it internally (or could just call the adapter API each time).

Reply:
We route the component `value` getter through the foundation.

Reply:
Whoops, I see. That's probably fair then, since both the foundation and the component need it.